server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/configuration/ConfigurationAggregate.java

```diff
@@ -0,0 +1,98 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.view.cassandra.configuration;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.eventsourcing.AggregateId;
+import org.apache.james.eventsourcing.Event;
+import org.apache.james.eventsourcing.eventstore.History;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+class ConfigurationAggregate {
+
+    private static class State {
+
+        private static State initial() {
+            return new State(Optional.empty());
+        }
+
+        private Optional<CassandraMailQueueViewConfiguration> maybeConfiguration;
+
+        State(Optional<CassandraMailQueueViewConfiguration> maybeConfiguration) {
+            Preconditions.checkNotNull(maybeConfiguration);
+            this.maybeConfiguration = maybeConfiguration;
+        }
+
+        void set(CassandraMailQueueViewConfiguration configuration) {
+            validateConfiguration(configuration);
```

Comment:
For each `configuration edited` event, I do the check inside the `State` change of `ConfigurationAggregate` before apply `configuration edited` to evensourcing

Reply:
There's a need for explanation here : State is what we call an "internal projection". Projections are views of system state that are built based on emitted events. An event is something that happened. You don't have any choice but to deal with it. So, when you plug everything together, it means that State can't take decision, it's aggregate's job.
And luckily, aggregate has a reference to State so it can take the decision.
Is it ok ?